core/services/chainlink/application.go

```diff
@@ -212,10 +209,31 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 
 		srvcs = append(srvcs, externalPeerWrapper)
 
+		networkSetup, err := capabilities.NewHardcodedDonNetworkSetup(externalPeerWrapper)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create hardcoded Don network setup: %w", err)
+		}
+
+		if opts.CapabilitiesRegistry == nil {
+			peerID := externalPeerWrapper.GetPeer().ID()
```

Comment:
This won't work. Peer is ready only after the wrapper is Start()-ed.

Reply:
PeerID is currently extracted from Keystore and that is not accessible until the services start.